tests/test_stackexchange.py

```diff
@@ -415,6 +415,22 @@ def request_callback(method, uri, headers):
         diff = after - before
         self.assertGreaterEqual(diff, 0.2)
 
+    def test_sanitize_for_archive(self):
+        """Test whether the sanitize method works properly"""
+
+        payload = {'order': 'desc',
+                   'site': 'stackoverflow',
+                   'sort': 'activity',
+                   'pagesize': 1,
+                   'key': 'aaa',
+                   'filter': 'Bf*y*ByQD_upZqozgU6lXL_62USGOoV3)MFNgiHqHpmO_Y-jHR',
+                   'page': 1,
+                   'tagged': 'python'}
+
+        url, headers, payload = StackExchangeClient.sanitize_for_archive(None, None, payload)
+        with self.assertRaises(KeyError):
+            payload.pop("key")
+
```

Comment:
I think you should check if `payload` contains the same keys and values minus `key` key. The method could be modifying the whole dictionary and this check won't catch that error.

This comment is applied to the rest of the backends which override this method.